File: android/app/src/main/java/com/flightmonitor/mobile/data/ShiftHandoverRepository.kt
package com.flightmonitor.mobile.data

import com.flightmonitor.mobile.api.ShiftHandoverApi
import com.flightmonitor.mobile.api.model.ShiftHandover
import com.flightmonitor.mobile.api.model.ShiftHandoverItem
import com.flightmonitor.mobile.api.model.ShiftHandoverItemAcknowledgeRequest

class ShiftHandoverRepository(
    private val api: ShiftHandoverApi,
) {
    suspend fun listHandovers(
        toUserId: String? = null,
        status: String? = null,
        limit: Int = 50,
        offset: Int = 0,
    ): List<ShiftHandover> {
        return api.listHandovers(
            toUserId = toUserId,
            status = status,
            limit = limit,
            offset = offset,
        )
    }

    suspend fun getHandover(handoverId: String): ShiftHandover {
        return api.getHandover(handoverId)
    }

    suspend fun acknowledgeItem(
        handoverId: String,
        itemId: String,
        acknowledged: Boolean = true,
    ): ShiftHandoverItem {
        return api.acknowledgeItem(
            handoverId = handoverId,
            itemId = itemId,
            payload = ShiftHandoverItemAcknowledgeRequest(acknowledged = acknowledged),
        )
    }

    suspend fun acknowledgeHandover(handoverId: String): ShiftHandover? {
        val response = api.acknowledgeHandover(handoverId = handoverId)
        return if (response.success) response.data else null
    }
}

File: services/api-server/crates/domain/src/ports/shift_handover_repository.rs
//! 交接班仓储 trait

use crate::error::DomainError;
use crate::models::shift_handover::{ShiftHandover, ShiftHandoverItem};
use async_trait::async_trait;
use chrono::NaiveDate;

#[async_trait]
pub trait ShiftHandoverRepository {
    async fn create(&self, handover: &ShiftHandover) -> Result<ShiftHandover, DomainError>;
    async fn find_by_id(&self, handover_id: &str) -> Result<Option<ShiftHandover>, DomainError>;
    async fn find_all(
        &self,
        shift_date: Option<NaiveDate>,
        shift_code: Option<&str>,
        status: Option<&str>,
        from_user_id: Option<&str>,
        to_user_id: Option<&str>,
        limit: i64,
        offset: i64,
    ) -> Result<Vec<ShiftHandover>, DomainError>;
    async fn submit(&self, handover_id: &str) -> Result<Option<ShiftHandover>, DomainError>;
    async fn acknowledge_item(
        &self,
        handover_id: &str,
        item_id: &str,
        acknowledged_by: &str,
        acknowledged: bool,
    ) -> Result<Option<ShiftHandoverItem>, DomainError>;
    async fn list_unacked_mandatory_titles(&self, handover_id: &str) -> Result<Vec<String>, DomainError>;
    async fn complete(
        &self,
        handover_id: &str,
        to_operator_name: Option<&str>,
        to_operator_job_title: Option<&str>,
    ) -> Result<Option<ShiftHandover>, DomainError>;
}
